Source/JavaScriptCore/inspector/remote/cocoa/RemoteInspectorCocoa.mm

```diff
@@ -346,6 +346,13 @@ static bool canAccessWebInspectorMachPort()
         pushListingsSoon();
 }
 
+void RemoteInspector::connectToWebInspector()
+{
+    dispatch_async(m_xpcQueue, ^{
```

Comment:
Is this being "async" going to continue the race, or is it somehow guaranteed to happen before the further lockdown point?

Reply:
That's a very good point! I initially thought that `setupXPCConnectionIfNeeded()` had to be called on `m_xpcQueue`, but looking closer, that does not seem to be the case. If `setupXPCConnectionIfNeeded()` can be called on the main thread, we should call it synchronously here.